spec/components/validated_field_component_spec.rb

```diff
@@ -39,7 +39,19 @@
     it 'renders with error message texts' do
       expect(rendered).to have_css(
         'script',
-        text: { valueMissing: t('forms.validation.required_checkbox') }.to_json,
+        text: { valueMissing: t('forms.validation.required_checkbox') }.to_json[1..-1],
```

Comment:
How come the `[1..-1]`... is that because the `"` gets escaped or something?

Reply:
It's because there may be multiple strings included in the JSON blob, e.g.

```json
{"valueMissing":"...","typeMismatch":"..."}
```

For the purpose of this test, since we care only about the `"valueMissing":"..."`, this basically takes what can be safely assumed as a single-key JSON object and trims off the `{` and `}` to get exactly `"valueMissing":"..."`, which would be part of a JSON object, regardless whether it has one or many keys.

I guess a (better?) alternative could be to parse it back out and assert against it as a Ruby hash.

Reply:
> I guess a (better?) alternative could be to parse it back out and assert against it as a Ruby hash.

I like that idea, it makes sense to me!

Reply:
Updated in c515dcee3.